### PR TITLE
Fixes clippy warnings

### PR DESCRIPTION
**Pull request by lmammino**

### CODE REVIEW

File: y2020/ex03/src/lib.rs [hunk 1]
```diff
@@ -1,10 +1,8 @@
 pub fn part1(_input: &str) -> u32 {
     let lines: Vec<Vec<char>> = _input
-            .lines()
-            .map(|s| {
-                s.chars().collect::<Vec<char>>()
-            })
-            .collect();
+        .lines()
+        .map(|s| s.chars().collect::<Vec<char>>())
+        .collect();
```
Comment (lmammino) [inline y2020/ex03/src/lib.rs:5]: just formatting

File: y2020/ex03/src/lib.rs [hunk 2]
```diff
@@ -19,18 +17,17 @@ pub fn part1(_input: &str) -> u32 {
...
+    let mut xs = vec![0; 5];
+    let ixs = vec![1, 3, 5, 7, 1];
+    let mut cnt = vec![0; 5];
     for (j, row) in lines.iter().enumerate() {
         for i in 0..xs.len() {
-            if i==4 && j%2==1 {  // jump the odd row when is the last rule (y must increment by 2)
-                continue
+            if i == 4 && j % 2 == 1 {
+                // jump the odd row when is the last rule (y must increment by 2)
+                continue;
```
Comment (lmammino) [inline y2020/ex03/src/lib.rs:30]: just formatting

File: y2020/ex03/src/lib.rs [hunk 3]
```diff
@@ -39,7 +36,7 @@ pub fn part2(_input: &str) -> u32 {
             xs[i] += ixs[i];
         }
     }
-    cnt.iter().fold(1, |a, b| a * b)
+    cnt.iter().product()
```
Comment (lmammino) [inline y2020/ex03/src/lib.rs:39]: this one was interesting to find out!

File: y2020/ex10/src/lib.rs [hunk 4]
```diff
@@ -132,11 +132,9 @@ pub fn part2(input: &str) -> u64 {
     // deals last implicit value (+3 implied at the end)
     combinations_by_chunk.push(tribonacci(elements_in_chunk as u32, cache));
 
-    let result: u64 = combinations_by_chunk
-        .iter()
-        .fold(1, |acc: u64, val| acc * (*val as u64));
+    let result: u64 = combinations_by_chunk.iter().map(|x| *x as u64).product();
```
Comment (lmammino) [inline y2020/ex10/src/lib.rs:135]: same here, i realised we could simply use `.product`

File: y2020/ex03/src/lib.rs (hunk 3 above)
Comment (eugenserbanescu) [reply y2020/ex03/src/lib.rs:39]: Ha! Cool!

File: y2020/ex06/src/lib.rs [hunk 5]
```diff
@@ -30,20 +30,21 @@ pub fn part2(input: &str) -> usize {
             answers += question_answers_set.len();
             question_answers_set.clear();
             is_first = true;
+        } else if is_first {
+            question_answers_set = line.chars().collect();
+            // for c in line.chars() {
+            //     question_answers_set.insert(c);
+            // }
+            is_first = false;
```
[the remark this answers is not in the document]
Comment (eugenserbanescu) [reply y2020/ex06/src/lib.rs:38]: By the way, why'd we leave commented code in?

[the remark this answers is not in the document]
Comment (lmammino) [reply y2020/ex06/src/lib.rs:38]: I think we wanted to leave it to remember what collect does, but i think now that we have consolidated that piece of knowledge it makes sense to remove it :D

**Pull request merged**